src/layouts/sidebar-sidecar/server.ts

```diff
@@ -27,11 +27,13 @@ export function getStaticGenerationFunctions({
   basePath,
   baseName,
   additionalRemarkPlugins = [],
+  scope,
 }: {
   product: Product
   basePath: string
   baseName: string
   additionalRemarkPlugins?: Pluggable[]
+  scope?: Record<string, $TSFixMe>
```

Comment:
Need to pass this through since [`/vagrant/docs/providers/vmware/vagrant-vmware-utility`](https://dev-portal-git-zsadd-vagrant-docs-views-hashicorp.vercel.app/vagrant/docs/providers/vmware/vagrant-vmware-utility) breaks the build if `VMWARE_UTILITY_VERSION` isn't present in `scope`.

See `Anything else?` section of PR description for details.

Reply:
I think scope is pretty freeform - I believe it could be basically anything depending on the product & page context. 

Great call to make that a bit more explicit - have added an optional `MdxScope` type argument for `getStaticGenerationFunctions`, and have used that in the `/vagrant/docs` route to pass the `{ VMWARE_UTIILITY_VERSION: string }` type we expect for MDX `scope` in the context of that specific page.

Let me know if that seems like a good solution 🙇‍♂️

> For posterity, I'd gone with `Record<string, $TSFixMe>` based on [the corresponding type declaration in `docs-page`](https://github.com/hashicorp/react-components/blob/0bca0d32dc7e4030b2237f3f397af49d6fd94a70/packages/docs-page/server/index.ts#L21).

Reply:
Looks like `Record<string, unknown>` is the way to go, I think I found the source? https://github.com/hashicorp/next-mdx-remote/blob/main/src/types.ts#L8